### PR TITLE
sub Q between types

### PR DESCRIPTION
**Pull request by 3pmile**
**Description**

This PR implements...
- [x] the Sub trait for Q - Z
- [x] the Sub trait for Q and Standard types u64, u32, u16, u8, i64, i32, i16, i8, f32, f64

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/rational/q/arithmetic/sub.rs [hunk 1]
```diff
@@ -108,3 +150,178 @@ mod test_sub {
...
+        let b: u64 = 1;
+        let c: u32 = 1;
+        let d: u16 = 1;
+        let e: u8 = 1;
+        let f: i64 = 1;
+        let g: i32 = 1;
+        let h: i16 = 1;
+        let i: i8 = 1;
+        let j: f32 = 1.3;
+        let k: f64 = 1.3;
```
Comment (MoogSven) [inline src/rational/q/arithmetic/sub.rs:237]: The variable names don't really explain what the value is about.
Maybe rename `a` to something like `rational` or `q_value`.
Similarly:
`b` -> `unsinged_int_64` or `uint_64`
`f` -> `int_64`
`j` -> `float_32`
etc.

**Review by MoogSven — APPROVED** [review]
Great work :+1: 

I would prefer it if the variable names in the last test would be more expressive than just alphabetic letters. This change should be relatively fast when using the rename feature of   VSCode (F2). However, applying it also to the other pull requests might take a bit of time.

File: src/rational/q/arithmetic/sub.rs (hunk 1 above)
Comment (3pmile) [reply src/rational/q/arithmetic/sub.rs:237]: discussed in group and current state is accepted

File: src/rational/q/arithmetic/sub.rs [hunk 2]
```diff
@@ -108,3 +150,178 @@ mod test_sub {
...
+    use std::str::FromStr;
+
+    /// testing subtraction for [`Q`] and [`Z`]
+    #[test]
+    fn sub() {
+        let a: Q = Q::from_str("5/7").unwrap();
+        let b: Z = Z::from(4);
+        let c: Q = a - b;
+        assert_eq!(c, Q::from_str("-23/7").unwrap());
+    }
```
[the remark this answers is not in the document]
Comment (3pmile) [reply src/rational/q/arithmetic/sub.rs:168]: discussed in group and current state is accepted

**Pull request merged**